### ISSUE TITLE
KeepNamesGeneratorStrategy generates wrong Keys.java

### ISSUE THREAD

**Issue by jeichhorn** [jOOQ/jOOQ#5783]
I am trying to generate jOOQ code for a legacy MySQL database which looks like the following:

```
create table TableA (
	Id bigint(20) auto_increment not null,
	Value varchar(20) not null,
	
	primary key (Id)
);

create table TableB (
	Id bigint(20) auto_increment not null,
	Value varchar(20) not null,
	TableAId bigint(20) not null,
	
	primary key (Id)
);

alter table TableB
add constraint FK_TABLE_B foreign key (TableAId) references TableA (Id);
```

I use maven to generate the jOOQ code:

```
<plugin>
	<groupId>org.jooq</groupId>
	<artifactId>jooq-codegen-maven</artifactId>
	<version>3.9.0</version>
	
	<executions>
		<execution>
			<phase>generate-sources</phase>
			<goals>
				<goal>generate</goal>
			</goals>
		</execution>
	</executions>

	<dependencies>
		<dependency>
			<groupId>mysql</groupId>
			<artifactId>mysql-connector-java</artifactId>
			<version>5.1.40</version>
		</dependency>
	</dependencies>
                    
	<configuration>
		<jdbc>
			<driver>com.mysql.jdbc.Driver</driver>
			<url>jdbc:mysql://localhost:3306/db</url>
			<user>username</user>
			<password>password</password>
		</jdbc>
		<generator>
			<database>
				<name>org.jooq.util.mysql.MySQLDatabase</name>

				<inputSchema>test</inputSchema>
				<outputSchema>test</outputSchema>
				<includes>.*</includes>
			</database>
			
			<target>
				<packageName>tests</packageName>
				<directory>target/generated-sources/jooq</directory>
			</target>

			<strategy>
				<name>org.jooq.util.KeepNamesGeneratorStrategy</name>
			</strategy>
		</generator>
	</configuration>
</plugin>
```
When using jOOQ 3.9.0 it generates the class `Keys` with identity entries like the following:
```
    public static final Identity<TableA, Long> IDENTITY_TableA = Identities0.IDENTITY_TableA;
    public static final Identity<TableB, Long> IDENTITY_TableB = Identities0.IDENTITY_TableB;

    private static class Identities0 extends AbstractKeys {
        public static Identity<TableA, Long> Id = createIdentity(tests.tables.TableA.TableA, tests.tables.TableA.TableA.Id);
        public static Identity<TableB, Long> Id = createIdentity(tests.tables.TableB.TableB, tests.tables.TableB.TableB.Id);
    }
```

When using jOOQ 3.8.6 the class `Keys`is generated without errors:

```
    public static final Identity<TableA, Long> IDENTITY_TableA = Identities0.IDENTITY_TableA;
    public static final Identity<TableB, Long> IDENTITY_TableB = Identities0.IDENTITY_TableB;

    private static class Identities0 extends AbstractKeys {
        public static Identity<TableA, Long> IDENTITY_TableA = createIdentity(tests.tables.TableA.TableA, tests.tables.TableA.TableA.Id);
        public static Identity<TableB, Long> IDENTITY_TableB = createIdentity(tests.tables.TableB.TableB, tests.tables.TableB.TableB.Id);
    }
```

**Comment by lukaseder** [comment]
Thanks for reporting, I can reproduce this issue. Will investigate immediately.

**Comment by lukaseder** [comment]
The regression was introduced with https://github.com/jOOQ/jOOQ/issues/4168 and https://github.com/jOOQ/jOOQ/commit/8c6a22505a81addb6a5c9fbe49c280735dbee6cd

**Comment by lukaseder** [comment]
#4168 wasn't merged to older releases, so only version 3.9.0 is affected

**Comment by lukaseder** [comment]
This is fixed for 3.10 and will be merged to 3.9.1 (https://github.com/jOOQ/jOOQ/issues/5786), which will hopefully be released by the end of the week.

Thanks again for reporting!
